WcaOnRails/app/assets/javascripts/currencies.js.erb

```diff
@@ -0,0 +1,60 @@
+window.wca = window.wca || {};
+<%
+  # Select currencies, for all the iso_code is matching the hash key.
+  # However for 2 currencies (JPY, GHS) there is a second currency with the same iso_code but a different key.
+  # They exist for backward compatibilities reason (see here: https://github.com/RubyMoney/money/blob/190683e7b84b66d11b83b7be32d019cf3c2fc114/config/currency_backwards_compatible.json), and we actually don't use them.
+  selected_currencies = Money::Currency.table.select { |k,v| k.to_s == v[:iso_code].downcase }
+  currencies_info_hash = Hash[selected_currencies.map do |k,v|
+    [v[:iso_code], {:symbol => v[:symbol],
+                    :symbol_first => v[:symbol_first],
+                    :subunit_to_unit => v[:subunit_to_unit]}]
+  end]
+%>
+wca._currenciesInfo =
+<%= raw(JSON.pretty_generate(currencies_info_hash, indent: '', object_nl: '', array_nl: '', space: '', space_after: '')) %>;
+
+wca.getCurrencyInfo = function(isoCode) {
+  return (isoCode in wca._currenciesInfo) ? wca._currenciesInfo[isoCode] : wca._currenciesInfo['USD'];
+}
+
+// Create a mask for an amount input in the given currency, using autoNumeric
+// 'action' can be either "init" or "update"
+// 'elemId' is a selector for the targeted input field
+wca.applyCurrencyMask = function(action, elemId, currencyIsoCode) {
+  var entry = wca.getCurrencyInfo(currencyIsoCode);
+  var currentVal = 0;
+
+  // Get current val
+  if (action == "update") {
+    currentVal = wca.getValueInCurrency(elemId);
+  } else if (action == "init") {
+    currentVal = $(elemId).val();
+  } else {
+    return;
+  }
+
+  // Reconfigure
+  $.data($(elemId)[0], "current_subunit_to_unit", entry.subunit_to_unit);
+  $(elemId).autoNumeric(action, {
+    currencySymbol: entry.symbol,
+    currencySymbolPlacement: (entry.symbol_first)?'p':'s',
+    decimalPlacesOverride: (entry.subunit_to_unit == 1) ? 0 : 2,
```

Comment:
Actually, the logic is more "unless the subunit_to_unit is 1, we need decimals".
There a few currency where the subunit is different; I'm not familiar with any real life use case, but it may be useful to permit people using a currency where the `subunit_to_unit` is 5 to input eg 2.40 to get a final amount of 12 in the lowest denomination.